src/testing_system/internal/adapter/retrievers/etl.py

```
@@ -0,0 +1,34 @@
+            self.chat_id = chat_id
+            self.k = k
+            self.clean = clean
+    
+    def retrieve(self, request: RetrievalRequest) -> RetrievalResponse:
+        params = {
+            "query": request.query,
+            "chat_id": self.chat_id,
+            "k": self.k,
+            "clean": self.clean
+        }
+        response = requests.get(f"{self.ETL_SERVICE_URL}/search", params=params)
```

Ruff (0.15.12)

[error] Probable use of `requests` call without timeout

(S113)